packages/block/src/header.ts

```diff
@@ -490,9 +489,13 @@ export class BlockHeader {
     return this._common.hardfork() || this._common.activeHardfork(this.number.toNumber())
   }
 
-  private async _getBlockByHash(blockchain: Blockchain, hash: Buffer): Promise<Block | undefined> {
+  private async _getHeaderByHash(
+    blockchain: Blockchain,
+    hash: Buffer
+  ): Promise<BlockHeader | undefined> {
     try {
-      return blockchain.getBlock(hash)
+      const header = (await blockchain.getBlock(hash)).header
+      return header
```

Comment:
Will comment here some notes:

In the original implementation the `blockchain.getBlock(hash)` Promise is **not** handled anywhere, it seems like it should `catch` the `NotFoundError` but it does not do this, because this `getBlock` call is **not** `await`ed. In this implementation this is fixed: in case that `getBlock` throws (for instance `NotFoundError`) then it will be handled by `catch`, as expected and probably intended in the first place.